strawberry_django/mutations/resolvers.py

```diff
@@ -656,11 +712,17 @@ def update_m2m(
                 data.pop(key_attr, None)
                 to_add.append(obj)
             elif data:
-                if key_attr not in data:
-                    manager.get_or_create(**data)
-                else:
-                    data.pop(key_attr)
-                    manager.create(**data)
+                # If we've reached here, the key_attr should be UNSET or missing. So
+                # let's remove it if it is there.
```

Comment:
Yep, fair question! This is probably the most contentious change - it's effectively dropping the fix for #360 (applied in #362).

The fix in #362 was made before `key_attr` was even added to the codebase, and I think the introduction of `key_attr` provides an alternate means to solve some of the cases that could trigger the issue in #360 (e.g. you are no longer forced to only lookup existing records by the `id` field, you could use the `name` field). I think it would have been valid to make the change I'm proposing here when `key_attr` was introduced.

That said, my change probably does break the specific example raised in #360 - which (I think...it's not 100% clear from the issue description) appears like it might want `key_attr` to be `id` for the parent, and `key_attr` to be `name` for the child. This currently isn't something you can do with strawberry-django.

Ultimately, I thought that it was a valid tradeoff to drop it though, given:

- The fix for #360 (#362) didn't include test cases and #360 doesn't really have enough information in it to retrospectively add them (we could make some guesses as to the missing information but we might be wrong)
- The fix for #360 seems like it might be wrong? Certainly I have cases in my own projects where I would always want to create new objects, not attempt to reuse existing ones, even if all input fields were the same. This seems to align with concerns you raised yourself a couple of months ago here: https://github.com/strawberry-graphql/strawberry-django/pull/604#discussion_r1703176752
- Given the subsequent introduction of `key_attr` (after #362) it seems like extending the `key_attr` functionality would be the best way to address #360 in a way that doesn't result in unexpected behaviour for other users.

What do you think?

Reply:
Yeah, I agree with your thoughts here! :)